api/api.go

```diff
@@ -503,30 +503,36 @@ func (b *BlockChainAPI) EstimateGas(
 		data = *args.Input
 	}
 
-	eoa := newEOATestAccount(b.config.COAKey.String()[2:])
-	// Provide a high enough gas for the tx to be able to execute
-	gas := uint64(15_000_000)
+	// provide a high enough gas for the tx to be able to execute
+	defaultGasLimit := uint64(15_000_000)
 	if args.Gas != nil {
-		gas = uint64(*args.Gas)
+		defaultGasLimit = uint64(*args.Gas)
 	}
 
-	txData := eoa.PrepareSignAndEncodeTx(
-		args.To,
+	txData, err := signGasEstimationTx(args.To,
 		data,
 		(*big.Int)(args.Value),
-		gas,
+		defaultGasLimit,
 		(*big.Int)(args.GasPrice),
 	)
+	if err != nil {
+		b.logger.Error().Err(err).Msg("failed to sign transaction for gas estimate")
+		return hexutil.Uint64(0), err
+	}
 
-	gas, err := b.evm.EstimateGas(ctx, txData)
+	estimatedGas, err := b.evm.EstimateGas(ctx, txData)
 	if err != nil {
 		b.logger.Error().Err(err).Msg("failed to estimate gas")
-		return hexutil.Uint64(0), err
+		return hexutil.Uint64(defaultGasLimit), err // return default gas limit
```

Comment:
Maybe return `nil` as an error then? As it might confuse clients, if there is an error, they won't use the value I guess.

Reply:
ohh, yeah you are right, missed it